packages/ui/src/components/FormRenderer/components/Checkbox/index.tsx

```diff
@@ -0,0 +1,136 @@
+            {...input}
+            {...option}
+            checked={(input.value || []).includes(option.value)}
+            controlId={`${input.name}-${option.value}`}
+            onBlur={() => input.onBlur()}
+            onFocus={() => input.onFocus()}
+            onChange={({ detail }) => {
+                if (detail.checked) {
+                    input.onChange([...input.value, option.value]);
+                } else {
+                    const newValue = (input.value || []).filter((v: string) => v !== option.value);
+                    input.onChange(newValue);
+                }
+            }}
+            data-testid={rest['data-testid'] ? `${rest['data-testid']}-${option.value}` : undefined}
+        >
+            {option.label}
+        </CheckboxComponent>
+    );
+};
+
+const Checkbox: FC<UseFieldApiConfig> = ({ component, ...props }) => {
+    const {
+        label,
+        description,
+        helperText,
+        info,
+        i18nStrings,
+        stretch,
+        secondaryControl,
+
+        options,
+        input,
+        isRequired,
+        isDisabled,
+        isReadOnly,
+
+        validateOnMount,
+        meta: { error, submitFailed },
+        showError,
+
+        ...rest
+    } = useFieldApi(props);
+    const controlId = useUniqueId(input.name);
+    const errorText = getErrorText(validateOnMount, submitFailed, showError, error);
+    if (options && options.length > 0) {
+        return (
+            <FormField
+                label={label}
+                description={description}
+                errorText={errorText}
+                constraintText={helperText}
+                info={info}
+                i18nStrings={i18nStrings}
+                stretch={stretch}
+                secondaryControl={secondaryControl}
+            >
+                <SpaceBetween direction="vertical" size="xs">
+                    {options.map((option: Option) => (
+                        <CheckboxMapping
+                            isDisabled={isDisabled}
+                            {...rest}
+                            input={input}
+                            option={option}
+                            key={option.value}
+                        />
+                    ))}
+                </SpaceBetween>
+            </FormField>
+        );
+    }
+
+    return (
+        <FormField
+            controlId={controlId}
+            errorText={errorText}
+            constraintText={helperText}
+            i18nStrings={i18nStrings}
+            stretch={stretch}
+            secondaryControl={secondaryControl}
+        >
+            <CheckboxComponent
+                {...rest}
+                {...input}
+                controlId={controlId}
+                name={input.name}
+                disabled={isDisabled}
+                checked={input.value || false}
+                description={description}
+                onBlur={() => input.onBlur()}
+                onFocus={() => input.onFocus()}
```

Comment:
```suggestion
                onBlur={input.onBlur}
                onFocus={input.onFocus}
```

Reply:
The type for both event handlers are having different signatures. This is to avoid type casting.